tests/Type/ArrayType/ArrayTypeTest.php

```diff
@@ -109,91 +109,152 @@ public function testContainsKeyIsNotStrict(
 	/**
 	 * @return mixed[][]|\Generator
 	 */
-	public function containsValueStrictDataProvider(): Generator
+	public function keyValueStrictDataProvider(): Generator
 	{
 		yield 'existing integer value' => [
 			'haystack' => [1, 2, 3],
+			'key' => 1,
 			'needle' => 2,
```

Comment:
I do not think we should use `needle` here - it is not mentioned in the cases, nor in the name. Basically `needle` is used in the implementation only in "search by value" context.

Reply:
ok, removed it.